File: sdks/dotnet/src/Resources/CategoriesResource.cs
using System.Threading.Tasks;
using Puxbay.SDK.Models;

namespace Puxbay.SDK.Resources
{
    public class CategoriesResource
    {
        private readonly PuxbayClient _client;
        public CategoriesResource(PuxbayClient client) { _client = client; }

        public async Task<PaginatedResponse<Category>> ListAsync(int page = 1)
        {
            return await _client.GetAsync<PaginatedResponse<Category>>($"categories/?page={page}");
        }

        public async Task<Category> GetAsync(string categoryId)
        {
            return await _client.GetAsync<Category>($"categories/{categoryId}/");
        }

        public async Task<Category> CreateAsync(Category category)
        {
            return await _client.PostAsync<Category>("categories/", category);
        }

        public async Task<Category> UpdateAsync(string categoryId, Category category)
        {
            return await _client.PatchAsync<Category>($"categories/{categoryId}/", category);
        }

        public async Task DeleteAsync(string categoryId)
        {
            await _client.DeleteAsync($"categories/{categoryId}/");
        }
    }
}

File: sdks/php/src/Resources/CategoriesResource.php
<?php

namespace Puxbay\Resources;

use Puxbay\Models\Category;
use Puxbay\Models\PaginatedResponse;

class CategoriesResource extends BaseResource
{
    public function list(int $page = 1): PaginatedResponse
    {
        $data = $this->get("categories/?page={$page}");
        return PaginatedResponse::fromArray($data, fn($item) => Category::fromArray($item));
    }

    public function getCategory(string $categoryId): Category
    {
        $data = $this->get("categories/{$categoryId}/");
        return Category::fromArray($data);
    }

    public function create(Category $category): Category
    {
        $data = $this->post('categories/', $category->toArray());
        return Category::fromArray($data);
    }

    public function update(string $categoryId, Category $category): Category
    {
        $data = $this->patch("categories/{$categoryId}/", $category->toArray());
        return Category::fromArray($data);
    }

    public function deleteCategory(string $categoryId): void
    {
        $this->delete("categories/{$categoryId}/");
    }
}
